rust/private/rustdoc_test.bzl

```diff
@@ -15,7 +15,71 @@
 """Rules for performing `rustdoc --test` on Bazel built crates"""
 
 load("//rust/private:common.bzl", "rust_common")
-load("//rust/private:utils.bzl", "dedent", "find_toolchain", "get_lib_name", "get_preferred_artifact")
+load("//rust/private:rustdoc.bzl", "rustdoc_compile_action")
+load("//rust/private:toolchain_utils.bzl", "find_sysroot")
+load("//rust/private:utils.bzl", "dedent", "find_toolchain")
+
+def _construct_writer_arguments(ctx, test_runner, action, crate_info, rust_toolchain):
```

Comment:
I'm kind of surprised at home much of this is actual logic vs re-calling functions defined elsewhere... What's unique about this that means that e.g. we need to set up the SYSROOT specially?

Reply:
This is largely solving for the fact that tests run from a runfiles directory where actions run in an execroot. But it also tracks what environment variables were explicitly added to the action. Happy to simplify if possible 😀

Reply:
Makes sense, and seems like it's a hard thing to share around - probably worth a comment to that effect :)